transport/http/outbound.go

```diff
@@ -495,7 +495,7 @@ func (o *Outbound) doWithPeer(
 ) (*http.Response, error) {
 	hreq.URL.Host = p.HostPort()
 
-	response, err := o.transport.client.Do(hreq.WithContext(ctx))
+	response, err := o.transport.client.Transport.RoundTrip(hreq.WithContext(ctx))
```

Comment:
If we don’t need the client, we could get rid of it entirely and just hold the transport directly.

Reply:
+1